src/nvim/change.c

```diff
@@ -523,19 +523,15 @@ void changed_lines_redraw_buf(buf_T *buf, linenr_T lnum, linenr_T lnume, linenr_
 {
   if (buf->b_mod_set) {
     // find the maximum area that must be redisplayed
-    if (lnum < buf->b_mod_top) {
-      buf->b_mod_top = lnum;
-    }
+    buf->b_mod_top = MIN(buf->b_mod_top, lnum);
     if (lnum < buf->b_mod_bot) {
       // adjust old bot position for xtra lines
       buf->b_mod_bot += xtra;
       if (buf->b_mod_bot < lnum) {
         buf->b_mod_bot = lnum;
       }
     }
-    if (lnume + xtra > buf->b_mod_bot) {
-      buf->b_mod_bot = lnume + xtra;
-    }
```

Comment:
This one, for example, breaks symmetry.

Reply:
Symmetry with what? Looks like I forgot to change line 530. Would that make it symmetrical again?

Reply:
Oh, I thought about it again. It's OK.